lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -329,6 +326,16 @@ void CGHeroInstance::initHero(CRandomGenerator & rand)
 		levelUpAutomatically(rand);
 	}
 
+	//load base hero bonuses, TODO: per-map loading of base hero bonuses
+	for(const auto & ob : VLC->modh->heroBaseBonuses)
+	{
+		auto bonus = ob;
+		bonus->source = Bonus::HERO_BASE_SKILL;
+		bonus->sid = id.getNum();
+		bonus->duration = Bonus::PERMANENT;
+		addNewBonus(bonus);
+	}
+
```

Comment:
This is to fix that mana issue?
I think we should use it only for mana (and anything else where per-hero level is needed), but keep rest of the bonuses in global node - bonuses in global node will also be able to affect non-hero nodes, e.g. towns without heroes. Useful functionality IMO

Reply:
@IvanSavenko there is now 2 possibilities to load global bonuses - from heroBaseBonuses and baseBonuses. It is only matter of Json to choose between loading global effects and base hero effects. I just move all hero bonuses to hero base bonuses. Should I move it back to json?

Reply:
Probably yes. Except for mana which has to be in hero I presume?

To act as hint for modders and to make sure that baseBonuses is in use & working

Reply:
@IvanSavenko I do not think than hero-only bonuses should affect hero only if he is inside global node. About baseBonuses example - I will add some unharming example.

Reply:
BaseBonuses is working, I tested it beforehand)

Reply:
Right now - yes. But what about in a month? In a year?
Don't want feature to die without us noticing that.

So I think we should prefer global node where possible, if only to reduce total amount of nodes in bonus system.
Mana should stay in per-hero list for now, since it breaks hero recruitment othervice, but everything else can go to global node.

Reply:
@IvanSavenko we should place bonuses than should affect unrecruited heroes into hero list.
For example, movement, mana limit, mana regen. But we can live with SIGHT_RADIUS, for example, in global node.